collect_app/src/main/java/org/odk/collect/android/widgets/utilities/RangeWidgetUtils.java

```diff
@@ -0,0 +1,118 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.TextView;
+
+import org.javarosa.core.model.RangeQuestion;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.R;
+import org.odk.collect.android.databinding.RangeWidgetHorizontalBinding;
+import org.odk.collect.android.databinding.RangeWidgetVerticalBinding;
+import org.odk.collect.android.utilities.ToastUtils;
+import org.odk.collect.android.views.CustomRangeSlider;
+
+import java.math.BigDecimal;
+
+public class RangeWidgetUtils {
+    private static final String VERTICAL_APPEARANCE = "vertical";
+
+    private RangeWidgetUtils() {
+    }
+
+    public static Object[] setUpLayoutElements(Context context, FormEntryPrompt prompt) {
+        View answerView;
+        CustomRangeSlider slider;
+        TextView currentValue;
+        TextView minValue;
+        TextView maxValue;
+
+        String appearance = prompt.getQuestion().getAppearanceAttr();
+
+        if (appearance != null && appearance.contains(VERTICAL_APPEARANCE)) {
+            RangeWidgetVerticalBinding rangeWidgetVerticalBinding = RangeWidgetVerticalBinding
+                    .inflate(((Activity) context).getLayoutInflater());
+            answerView = rangeWidgetVerticalBinding.getRoot();
+
+            slider = rangeWidgetVerticalBinding.slider;
+            currentValue = rangeWidgetVerticalBinding.currentValue;
+            minValue = rangeWidgetVerticalBinding.minValue;
+            maxValue = rangeWidgetVerticalBinding.maxValue;
+        } else {
+            RangeWidgetHorizontalBinding rangeWidgetHorizontalBinding = RangeWidgetHorizontalBinding
+                    .inflate(((Activity) context).getLayoutInflater());
+            answerView = rangeWidgetHorizontalBinding.getRoot();
+
+            slider = rangeWidgetHorizontalBinding.slider;
+            currentValue = rangeWidgetHorizontalBinding.currentValue;
+            minValue = rangeWidgetHorizontalBinding.minValue;
+            maxValue = rangeWidgetHorizontalBinding.maxValue;
+        }
+        setUpWidgetParameters((RangeQuestion) prompt.getQuestion(), minValue, maxValue);
+
+        if (prompt.isReadOnly()) {
+            slider.setEnabled(false);
+        }
+
+        return new Object[] {answerView, slider, currentValue};
```

Comment:
This should probably just return a custom class rather than an array. You could nest it in here and it could be called something like `RangeWidgetLayoutElements`?